shared/hybrid/AppDelegate+SalesforceHybridSDK.m

```diff
@@ -92,9 +92,15 @@ - (BOOL)sfsdk_swizzled_application:(UIApplication *)application didFinishLaunchi
     
     [self initializeAppViewState];
      __weak __typeof (self) weakSelf = self;
-    [SFSDKAuthHelper loginIfRequired:^{
-        [weakSelf setupRootViewController];
-    }];
+
+    if (SalesforceHybridSDKManager.sharedManager.appConfig.shouldAuthenticate) {
+        [SFSDKAuthHelper loginIfRequired:^{
```

Comment:
@trooper2013 @wmathurin Just out of curiosity, why's it called `loginIfRequired` if he doesn't take into account `shouldAuthenticate`?

Reply:
Raj and I discussed that.
Right now it's does a login flow if not already logged in.
But the name could be assumed to mean login if required by the app config.
We could modify the internals of loginIfRequired to take into account shouldAuthenticate. Not sure if that might surprise some customers since it's a behavior change (even though it seems consistent with the method name).

Reply:
Makes sense. We can leave it as-is.

Reply:
@trooper2013 We also need to fix the native templates (see https://github.com/forcedotcom/SalesforceMobileSDK-Templates/issues/179) or maybe we should change loginIfRequired to check the shouldAuthenticate flag.